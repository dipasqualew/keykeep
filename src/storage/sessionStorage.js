
import StorageHandler from './handler';

/**
 * SessionStorage Handler
 */
export default class SessionStorageHandler extends StorageHandler {

    /**
     * Returns the SessionStorage instance.
     *
     * @returns {WindowSessionStorage}
     */
    get ss() {
        return this.context.sessionStorage;
    }

    /**
     * @inheritdoc
     */
    async implSave(context, data) {
        this.ss.setItem(this.key, data);
    }

    /**
     * @inheritdoc
     */
    implLoad() {
        return this.ss.getItem(this.key);
    }

}
