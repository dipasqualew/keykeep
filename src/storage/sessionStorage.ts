/**
 * @module Storage/SessionStorageHandler
 */


import { EncryptionContext } from '../strategy/handler';
import { StorageHandler, Serialized } from './handler';

/**
 * SessionStorage Handler
 */
export class SessionStorageHandler extends StorageHandler {

  static serializer = 'SessionStorageHandler';

  /**
     * Returns the SessionStorage instance.
     */
  get ss(): Storage {
    return window.sessionStorage;
  }

  async implSave(_context: EncryptionContext, serialized: Serialized): Promise<void> {
    const raw = JSON.stringify(serialized);
    this.ss.setItem(this.key, raw);
  }

  async implLoad(_context: EncryptionContext): Promise<Serialized | null> {
    const raw = this.ss.getItem(this.key);

    if (raw) {
      return JSON.parse(raw);
    }

    return null;
  }

  async implClear(_context: EncryptionContext): Promise<void> {
    this.ss.removeItem(this.key);
  }

}
