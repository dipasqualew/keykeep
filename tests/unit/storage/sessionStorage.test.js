import sessionStorage from '../../../src/storage/sessionStorage';
import { storageHandlerSuite } from './mixins';

storageHandlerSuite(sessionStorage);
