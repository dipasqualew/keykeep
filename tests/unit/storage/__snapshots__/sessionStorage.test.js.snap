// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serialize Serializes the payload 1`] = `
Object {
  "meta": Object {
    "datetime": Any<Number>,
    "key": "StorageHandlerKey",
    "serializer": "SessionStorageHandler",
    "version": 1,
  },
  "payload": "encrypted",
}
`;
